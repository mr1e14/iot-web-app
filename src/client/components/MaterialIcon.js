import React from "react";
import "../css/app.css";
import "../css/material-ui-icons.css";
import Icon from "@material-ui/core/Icon";
import matches from "../mediaQuery";

const MaterialIcon = props => {
  const iconSize = matches("xs") ? "md-48" : "md-64";
  return <Icon classes={{ root: iconSize }}>{props.iconName}</Icon>;
};

export default MaterialIcon;
