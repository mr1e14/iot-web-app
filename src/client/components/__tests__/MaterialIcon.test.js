import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

jest.mock("../../mediaQuery");

import MaterialIcon from "../MaterialIcon";
import matches from "../../mediaQuery";

describe("MaterialIcon", () => {
  let tree;
  let component;
  describe("on xs devices", () => {
    beforeEach(() => {
      matches.mockReturnValue(true);
      component = renderer.create(<MaterialIcon iconName="account_circle" />);
      tree = component.toJSON();
    });
    it("renders correctly", () => {
      expect(tree).toMatchSnapshot();
    });
  });
  describe("on larger devices", () => {
    beforeEach(() => {
      matches.mockReturnValue(false);
      component = renderer.create(<MaterialIcon iconName="account_circle" />);
      tree = component.toJSON();
    });
    it("renders correctly", () => {
      expect(tree).toMatchSnapshot();
    });
  });
  describe("size changes based on screen size", () => {
    describe("On xs devices", () => {
      beforeEach(() => {
        matches.mockReturnValue(true);
      });
      it("has .md-48 class", () => {
        const wrapper = mount(<MaterialIcon iconName="account_circle" />);
        expect(wrapper.find(".material-icons.md-48").exists()).toEqual(true);
      });
    });
    describe("On larger devices", () => {
      beforeEach(() => {
        matches.mockReturnValue(false);
      });
      it("has .md-64 class", () => {
        const wrapper = mount(<MaterialIcon iconName="account_circle" />);
        expect(wrapper.find(".material-icons.md-64").exists()).toEqual(true);
      });
    });
  });
});
