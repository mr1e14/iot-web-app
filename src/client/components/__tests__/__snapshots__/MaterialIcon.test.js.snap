// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaterialIcon on larger devices renders correctly 1`] = `
<span
  aria-hidden="true"
  className="material-icons MuiIcon-root-1 md-64"
>
  account_circle
</span>
`;

exports[`MaterialIcon on xs devices renders correctly 1`] = `
<span
  aria-hidden="true"
  className="material-icons MuiIcon-root-1 md-48"
>
  account_circle
</span>
`;
